### ISSUE TITLE
Fix selected search filter label regression

### ISSUE THREAD

**Issue by gsidebo** [mitodl/micromasters#3037]
Selected search filter labels were fixed in issue #3006, but the structure of the index was being changed in #2942 at the same time. As a result, #2942 was merged without fixing the selected search filter labels that had regressed. It should be pretty straightforward to fix these again. If there's a way for us to either test for these labels or to solve this problem in a general way that doesn't require us to change a bunch of constants, we really should do that. As it is, it is incredibly easy for this to regress again

![ss 2017-04-10 at 13 43 03](https://cloud.githubusercontent.com/assets/14932219/24875060/d227066e-1df4-11e7-8555-7ba44654c378.png)

**Comment by amir-qayyum-khan** [comment]
it is working fine on MM local instance and master branch. My super admin and staff access at rc is revoked, could not able to see behaviour there.

<img width="954" alt="screen shot 2017-04-11 at 1 10 36 pm" src="https://cloud.githubusercontent.com/assets/10431250/24899100/8ccd17e2-1eb8-11e7-938f-e725035b0fd7.png">

**Comment by amir-qayyum-khan** [comment]
@pdpinch 
what i did is build, migrate and reindex latest master and this was working fine
